demo-vue/src/mixins/vueStricklandMixinFactory.js

```diff
@@ -0,0 +1,178 @@
+import validate from 'strickland';
+
+const TAGNAME_SELECT = 'SELECT';
+const TAGNAME_INPUT = 'INPUT';
+const ATTRIBUTE_TYPE_CHECKBOX = 'checkbox';
+const ATTRIBUTE_TYPE_RADIO = 'radio';
+const ATTRIBUTE_MULTIPLE = 'multiple';
+
+const isSelectInput = (element) => element.tagName === TAGNAME_SELECT;
+const isCheckboxInput = (element) => element.tagName === TAGNAME_INPUT && element.type === ATTRIBUTE_TYPE_CHECKBOX;
+const isRadioInput = (element) => element.tagName === TAGNAME_INPUT && element.type === ATTRIBUTE_TYPE_RADIO;
+const shouldUseChangeEvent = (element) => isSelectInput(element) || isCheckboxInput(element) || isRadioInput(element);
+const shouldUseChangeEventOnDelay = (element) => isSelectInput(element) && element.attributes[ATTRIBUTE_MULTIPLE];
+
+export default (validator) => ({
+  data: () => ({
+    validator,
+    validation: validator.emptyResults()
+  }),
+  computed: {
+    isValid: () => this.validation && this.validation.isValid
+  },
+  methods: {
+    validateAsync (fieldResult, fieldName) {
+      this.logValidation(`strickland:validateAsync method triggered for field: '${fieldName}'.`);
+      fieldResult.validateAsync(() => this.form[fieldName])
+        .then((result) => {
+          // TODO: Do we need to check for change/race condition as per:
+          // https://strickland.io/docs/Async/RaceConditions.html
+          this.validation = this.validator.updateFieldResults(this.validation, { [fieldName]: result });
+        })
+        .catch((result) => {
+          console.error(`Error on async validation for field: ${fieldName}.`, result);
+        });
+    },
+    getDependentsNeedingRevalidation (fieldName, validationResults) {
+      let dependents = this.validationDependencies ? this.validationDependencies[fieldName] : null;
+      return dependents
+        ? Object.keys(validationResults).filter((field) => dependents.includes(field))
+        : [];
+    },
+    stricklandOnChange (event, sender) {
```

Comment:
Per the feedback above, I recommend renaming this to `validateOnChange`.

Reply:
This is mostly for namespacing. With mixins, the properties on the mixin object are merged into the component "inheriting" the mixin. So I'm just trying to avoid conflicts when a user's component defined their own functions with the same name. There's no native support for namespacing and it's been rejected by vue core team, but I could group them under a computed property which has the same affect. 

Again, mostly for when I move this into a plugin, so will leave for now and we can discuss later about the namespace: `vueStrickland.Xyz` or `somethingElse.Xyz`.